### ISSUE TITLE
Wider support for regression families in brms

### ISSUE THREAD

**Issue by StaffanBetner** [rvlenth/emmeans#195]
I have found that when using a hurdle_gamma model from brms in emmeans it only uses the model for the positive part of the model. I think this could be fixed if the pp_expect function is used to get the posterior samples of expectations.

**Comment by rvlenth** [comment]
There are actually several features of brms models that are not supported by **emmeans**. And the problem is that those models offer a lot of possibilities and I lack the needed expertise on that package's capabilities.

We have had some communication about this -- see paul-buerkner/brms#418 for example. I think the best solution would be for me to work toward moving the emmeans support for brms models completely out of **emmeans** and into **brms**. I'll see if I can put together a pull request for **brms** for that purpose, with the existing code. Then Paul can take it from there as he is the one who would know how best to enhance this support.

**Comment by rvlenth** [comment]
Update. I have contributed some code to the brms package, and it has been merged-in and extended somewhat. That code is not yet in the master branch of brms, as it needs more validation. There were several needed changes even in my contributed code to cover some functions that have been deprecated. But I think we are making some progress on getting this to work for at least quite a few more models. I haven't had a chance to test a hurdle model yet.
